### DISCUSSION TITLE
Internal Error when using product filters

### DISCUSSION THREAD

**Discussion by code-by-gijs** [Shopify/storefront-api-feedback#121] [Help] [answered]
Hi!

I've been using the Shopify Storefront API for a while now. Since 2022-01 I've used the new product filters. They used to work fine. As of the release of 2022-04 we have been using this version. But since about two weeks we're experiencing problems when using the product filters on two separate environments. The first environment started throwing errors two weeks ago while the other environment was still working fine. Last week they where both throwing erros and since this week only the second environment throws errors. The errors only occour when enableing a filter trough the productFilter in the query variables. When no filters are used and all products are fetched, the query works fine. Also, when changing just one field in the query (removing an id field for example), the query works fine! :-(

I've been trying to contact Shopify but without any luck. They only point towards the community forums. I've created a post on the community forums but there was no response. I'm really hoping someone on this discussion board can point me in the right direction because I'm really lost at this point! 

Thanks in advance,

Gijs!

```
query CollectionByHandle($handle: String!, $productLimit: Int, $productFilters: [ProductFilter!], $productSortKey: ProductCollectionSortKeys, $productCursor: String) {
  availableProductFilters: collection(handle: $handle) {
    products(first: 0) {
      filters {
        id
        label
        type
        values {
          id
          count
          input
          label
        }
      }
    }
  }
  collection(handle: $handle) {
    ...CollectionMinimal
    products(
      first: $productLimit
      filters: $productFilters
      sortKey: $productSortKey
      after: $productCursor
    ) {
      pageInfo {
        hasNextPage
        hasPreviousPage
      }
      edges {
        cursor
        node {
          ...ProductMinimal
          images(first: 10) {
            edges {
              node {
                id
                url
                altText
              }
            }
          }
          variants(first: 25) {
            edges {
              node {
                ...Variant
              }
            }
          }
        }
      }
    }
  }
}

fragment CollectionMinimal on Collection {
  id
  title
  handle
  parentId: metafield(namespace: "extra_fields", key: "parent_collection") {
    value
  }
  image: metafield(namespace: "extra_fields", key: "extra_image") {
    value
  }
  description: metafield(namespace: "extra_fields", key: "extra_content") {
    value
  }
}

fragment ProductMinimal on Product {
  id
  handle
  title
  vendor
  productType
  tags
  descriptionHtml
  compareAtPriceRange {
    minVariantPrice {
      amount
      currencyCode
    }
    maxVariantPrice {
      amount
      currencyCode
    }
  }
  priceRange {
    minVariantPrice {
      amount
      currencyCode
    }
    maxVariantPrice {
      amount
      currencyCode
    }
  }
  productOrderType: metafield(namespace: "extra_fields", key: "product_order_type") {
    key
    value
  }
  usps: metafield(namespace: "extra_fields", key: "usps") {
    key
    value
  }
  orderLimit: metafield(namespace: "extra_fields", key: "order_limit") {
    key
    value
  }
  showShirtsChart: metafield(namespace: "extra_fields", key: "show_shirts_chart") {
    key
    value
  }
  label: metafield(namespace: "extra_fields", key: "label") {
    key
    value
  }
  warranty: metafield(namespace: "extra_fields", key: "warranty") {
    key
    value
  }
  descriptionExt: metafield(namespace: "extra_fields", key: "description") {
    key
    value
  }
  specifications: metafield(namespace: "extra_fields", key: "specifications") {
    key
    value
  }
  serviceQuestions: metafield(namespace: "extra_fields", key: "service_questions") {
    key
    value
  }
  preOrderDate: metafield(namespace: "extra_fields", key: "pre_order_date") {
    key
    value
  }
}

fragment Variant on ProductVariant {
  id
  title
  sku
  image {
    id
    altText
    url
  }
  priceV2 {
    amount
    currencyCode
  }
  compareAtPriceV2 {
    amount
    currencyCode
  }
  product {
    ...ProductMinimal
  }
  quantityAvailable
  availableForSale
}
```

Variables
```
{
    "handle": "my-collection-handle",
    "productCursor": null,
    "productFilters": [
        {
            "productVendor": "Primary"
        }
    ],
    "productLimit": 24,
    "productSortKey": "COLLECTION_DEFAULT"
}
```

The error
```
{
    "errors": [
        {
            "message": "Internal error. Looks like something went wrong on our end.\nRequest ID: 07a7f70b-a8e0-48f7-a2fd-16b30c9c6473 (include this in support requests).",
            "extensions": {
                "code": "INTERNAL_SERVER_ERROR",
                "requestId": "07a7f70b-a8e0-48f7-a2fd-16b30c9c6473"
            }
        }
    ]
}
```

**Comment by johncraigcole** [comment, marked as answer]
Hey @code-by-gijs sorry no one has followed up yet with you on this. Some time has elapsed since you asked the question and was wondering if you would mind running the query again and providing me with a new request-id as the one provided originally isn't showing me a lot of details which is possibly related to our log retention window.

**Reply by code-by-gijs** [reply]
Hi @johncraigcole ! Thank you for your response! This is one of the errors we've been having today:

```
{"errors":[{"message":"Internal error. Looks like something went wrong on our end.\nRequest ID: 28f25104-c6c6-4d41-8d45-8417fbde4220 (include this in support requests).","extensions":{"code":"INTERNAL_SERVER_ERROR","requestId":"28f25104-c6c6-4d41-8d45-8417fbde4220"}}]}
```

**Reply by johncraigcole** [reply]
Thanks for the request-id. I'll take a look now.

**Reply by johncraigcole** [reply]
Hey @code-by-gijs can you let me know if you run the same query right now if it still is happening. Just chatting with the engineering team and there was a fixed pushed fairly recently.

**Reply by code-by-gijs** [reply]
Hey @johncraigcole , it seems to be working again! Was the fix pushed to all api versions? Do you know what actually went wrong?

**Reply by johncraigcole** [reply]
Hey @code-by-gijs we're always looking at ways to improve response times and move these API queries out to the edge and as close to the api consumer as possible. The faceted filtering query just happened to be a query type that was presenting some challenges in our efforts to make some long term improvements on performance. We've pushed a fixed to production that allows these query types to work as they were before while we sort out the underlying issue. The fix should be applicable to all API versions.

**Reply by code-by-gijs** [reply]
Thanks for the explanation! I'm very happy this fixed our issues!
